src/server/snapshot.cc

```diff
@@ -69,7 +72,19 @@ void SliceSnapshot::Start(bool stream_journal, const Cancellation* cll) {
     journal_cb_id_ = journal->RegisterOnChange(std::move(journal_cb));
   }
 
-  serializer_ = std::make_unique<RdbSerializer>(compression_mode_);
+  const auto flush_threshold = absl::GetFlag(FLAGS_flush_big_entries_threshold);
+  std::function<bool(size_t)> flush_fun;
+  if (flush_threshold != 0 && allow_flush_fun) {
+    flush_fun = [this, flush_threshold](size_t bytes_serialized) {
+      if (bytes_serialized > flush_threshold) {
+        auto serialized = Serialize();
```

Comment:
I think we also found a bug. We can't send `delayed entries` because it could be that we are in a middle of a chunk (and if we serialize those delayed entries we just interleaved the channel breaking our invariants). The same thing applies for `PushSerializedToChannel` (and I patched this accordingly now)